File: rag-backend/test_qdrant_connection.js
require('dotenv').config({ path: require('path').resolve(__dirname, '../.env') });
const { QdrantClient } = require('@qdrant/js-client-rest');

const QDRANT_URL = process.env.QDRANT_URL;
const QDRANT_API_KEY = process.env.QDRANT_API_KEY;

console.log('QDRANT_URL:', QDRANT_URL);
console.log('Attempting connection to Qdrant...');

const qdrantClient = new QdrantClient({
    url: QDRANT_URL,
    apiKey: QDRANT_API_KEY,
});

async function testConnection() {
    try {
        console.log('Testing connection...');
        const collections = await qdrantClient.getCollections();
        console.log('Connected successfully!');
        console.log('Number of collections:', collections.collections.length);
        console.log('Collections:', collections.collections.map(c => c.name));

        const targetCollection = collections.collections.find(col => col.name === 'docusaurus_rag');
        if (targetCollection) {
            const collectionInfo = await qdrantClient.getCollection('docusaurus_rag');
            console.log(`\ndocusaurus_rag collection exists!`);
            console.log(`Points count: ${collectionInfo.points_count}`);
        } else {
            console.log('\ndocusaurus_rag collection does not exist yet.');
        }
    } catch (error) {
        console.error('Connection failed:', error.message);
        console.error('Error details:', error);
    }
}

testConnection();

// require('dotenv').config();
// const { QdrantClient } = require('@qdrant/js-client-rest');

// // Use the values from your .env file
// const QDRANT_URL = process.env.QDRANT_URL;
// const QDRANT_API_KEY = process.env.QDRANT_API_KEY;

// console.log('QDRANT_URL:', QDRANT_URL);
// console.log('Attempting connection to Qdrant...');

// const qdrantClient = new QdrantClient({
//     url: QDRANT_URL,  // Note: using 'url' instead of 'host' for cloud instances
//     apiKey: QDRANT_API_KEY,
// });

// async function testConnection() {
//     try {
//         console.log('Testing connection...');
//         const collections = await qdrantClient.getCollections();
//         console.log('Connected successfully!');
//         console.log('Number of collections:', collections.collections.length);
//         console.log('Collections:', collections.collections.map(c => c.name));

//         // Check specifically for the docusaurus_rag collection
//         const targetCollection = collections.collections.find(col => col.name === 'docusaurus_rag');
//         if (targetCollection) {
//             const collectionInfo = await qdrantClient.getCollection('docusaurus_rag');
//             console.log(`\ndocusaurus_rag collection exists!`);
//             console.log(`Points count: ${collectionInfo.points_count}`);
//         } else {
//             console.log('\ndocusaurus_rag collection does not exist yet.');
//         }
//     } catch (error) {
//         console.error('Connection failed:', error.message);
//         console.error('Error details:', error);
//     }
// }

// testConnection();

File: test_qdrant_connection.py
import asyncio
import os
from qdrant_client import QdrantClient

# Load environment variables
from dotenv import load_dotenv
load_dotenv()

# Configuration
QDRANT_URL = os.getenv("QDRANT_URL", "http://localhost:6333")
QDRANT_API_KEY = os.getenv("QDRANT_API_KEY")
COLLECTION_NAME = "docusaurus-rag"

def test_qdrant_connection():
    """Test the Qdrant connection directly"""
    try:
        # Initialize Qdrant client
        if QDRANT_API_KEY:
            qdrant_client = QdrantClient(
                url=QDRANT_URL,
                api_key=QDRANT_API_KEY,
            )
        else:
            qdrant_client = QdrantClient(
                host='localhost',
                port=6333
            )

        print(f"Testing connection to Qdrant at: {QDRANT_URL}")
        print(f"Collection name: {COLLECTION_NAME}")

        # Test connection and get collection info
        collection_info = qdrant_client.get_collection(COLLECTION_NAME)

        print("SUCCESS: Connection successful!")
        print(f"Collection status: available")
        print(f"Vector count: {collection_info.points_count}")

        # Try to get vector dimensions
        vector_dimensions = None
        if hasattr(collection_info, 'config') and collection_info.config:
            if hasattr(collection_info.config, 'params') and collection_info.config.params:
                vector_config = collection_info.config.params.vectors
                if hasattr(vector_config, 'size'):
                    vector_dimensions = vector_config.size
                elif isinstance(vector_config, dict) and 'size' in vector_config:
                    vector_dimensions = vector_config['size']

        print(f"Vector dimensions: {vector_dimensions}")

        return True

    except Exception as e:
        print(f"ERROR: Connection failed: {str(e)}")
        return False

if __name__ == "__main__":
    test_qdrant_connection()